chopper/pubspec.yaml

```diff
@@ -1,28 +1,29 @@
 name: chopper
-description: Chopper is an http client generator using source_gen and inspired from retrofit
-version: 3.0.6
-homepage: https://hadrien-lejard.gitbook.io/chopper
+description: Chopper is an http client generator using source_gen, inspired by Retrofit
+version: 4.0.0-nullsafety.0
+documentation: https://hadrien-lejard.gitbook.io/chopper
 repository: https://github.com/lejard-h/chopper
 author: Hadrien Lejard <hadrien.lejard@gmail.com>
 
 environment:
-  sdk: ">=2.1.0 <3.0.0"
+  sdk: ">=2.12.0 <3.0.0"
```

Comment:
I think this is okay for the _final_ version 4.0.0, but shouldn't this be `">=2.12.0-0 <3.0.0"` during the `4.0.0-nullsafety.x` development versions so people can already use this in projects which are not migrated to the latest stable yet?

Reply:
I can change that yes

Reply:
@jpelgrim unfortunately I need to do a lot dependency changes to support unstable sdk (source_gen, build, http ...)
We will support stable 2.12 only sorry :/

Reply:
Okay, understand. Maybe rename the PR then?

Reply:
@jpelgrim has a point here. I know we discussed in an earlier thread that we want to test this release out in the wild before saying that it is stable, but considering the above, we just might "lie" to our users ("unstable" release with stable dependencies and requirements).

Reply:
I don't think it's a problem, using `4.0.0-nullsafety.x` just mean the package itself is unstable for any reasons (partial migration, untested ...) it does not mean we use unstable sdk or unstable dependencies (it could but not here).

If the naming is the issue, we could just rename it to `4.0.0-dev.x`

Reply:
I just want to avoid the release of a stable `4.0.0` then introduce breaking change a few weeks later because of an issue on null safety.

Reply:
4.0.0-nullsafety.x is a really good well-known approach

Reply:
Yeah, I think so too. The argument was

> Unfortunately I need to do a lot dependency changes to support unstable sdk (source_gen, build, http ...)

But I think that needs to be done for the final null-safe version anyway. So if we just depend on the latest `<package>-nullsafety.x` version for all the dependencies in the chopper package and depend on Dart SDK `">=2.12.0-0 <3.0.0"`  is a good approach to support a set of true `4.0.0-nullsafety.x` versions people can play around with.